code/modules/atmospherics/machinery/pipes/pipes.dm

```diff
@@ -73,25 +97,33 @@
 /obj/machinery/atmospherics/pipe/return_pipenet()
 	return parent
 
-/obj/machinery/atmospherics/pipe/set_pipenet(datum/pipeline/pipenet_to_set)
-	parent = pipenet_to_set
+/obj/machinery/atmospherics/pipe/replace_pipenet(datum/pipeline/old_pipenet, datum/pipeline/new_pipenet)
+	if(parent && has_gas_visuals)
+		vis_contents -= parent.GetGasVisual('icons/obj/pipes_n_cables/!pipe_gas_overlays.dmi')
 
-/obj/machinery/atmospherics/pipe/Destroy()
-	QDEL_NULL(parent)
+	parent = new_pipenet
 
-	releaseAirToTurf()
+	if(parent && has_gas_visuals) // null is a valid argument here
+		vis_contents += parent.GetGasVisual('icons/obj/pipes_n_cables/!pipe_gas_overlays.dmi')
 
-	var/turf/local_turf = loc
-	for(var/obj/machinery/meter/meter in local_turf)
-		if(meter.target != src)
-			continue
-		var/obj/item/pipe_meter/meter_object = new (local_turf)
-		meter.transfer_fingerprints_to(meter_object)
-		qdel(meter)
+/obj/machinery/atmospherics/pipe/return_pipenets()
+	. = list(parent)
+
+////////////////////
+// APPEARANCE STUFF
+
+/obj/machinery/atmospherics/pipe/update_icon()
+	update_pipe_icon()
+	update_layer()
 	return ..()
 
 /obj/machinery/atmospherics/pipe/proc/update_pipe_icon()
-	icon = 'icons/obj/pipes_n_cables/pipes_bitmask.dmi'
+	switch(initialize_directions)
+		if(NORTH, EAST, SOUTH, WEST) // Pipes with only a single connection aren't handled by this system
```

Comment:
what about multiz pipes?

Reply:
In short. the pipe icon file never had states for pipes with only a single connected dir. This is specifically to fix multi-z pipes as previously they ran this code but the icon they grabbed as a result was invalid. The icon file now has an error sprite in the blank icon state which meant that multi-z pipes were grabbing the error sprite.